### PR TITLE
chore: finesse changeset config

### PR DESCRIPTION
**Pull request by andyhass**
With these changes, the workflow for making changes to the Elements repo and publishing those changes to NPM are as follows:

1. Build feature or make change
2. Run `pnpm bump-version` to update the version number and changelog.
3. Create PR with these changes**
4. Merge PR.  This will automatically publish a new version to NPM.

**☝️ Note: If Step 2 is skipped, the CI pipeline for the PR will be in a failure state.  An escape hatch exists for this check (e.g. if we're building a feature across multiple PRs), and it can be skipped by adding `[skip-version-check]` to the PR title.

### CODE REVIEW

File: .github/workflows/package-version-check.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+name: Version Update Check
+
+on:
+  pull_request:
+    branches:
+      - main
+    paths:
```
Comment (andyhass) [inline .github/workflows/package-version-check.yml:7]: 👇 If changes were made to files in `src` but the package.json's version hasn't been updated, the CI pipeline will be red.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -29,7 +29,7 @@ jobs:
       - name: Install dependencies
         run: pnpm install --frozen-lockfile
```
Comment (andyhass) [inline .github/workflows/release.yml:31]: 👇 Changing this name to be more appropriate - it now just pushes to NPM.

File: package.json [hunk 3]
```diff
@@ -16,7 +16,8 @@
   "scripts": {
     "test": "echo \"Error: no test specified\" && exit 1",
     "build": "rollup --config",
-    "release": "pnpm build && changeset publish --dry-run"
```
Comment (andyhass) [inline package.json:19]: 👇 The `bump-version` command will guide the dev through the process of creating a changelog and updating the package.json version. 

https://github.com/user-attachments/assets/33bc3497-c85c-4123-9d6c-82686ba12917

**Review by JesseKuntz — APPROVED** [review]
🔥

**Pull request merged**